core/crypto/client_tcert_pool_mt.go

```diff
@@ -63,7 +63,26 @@ func (tCertPool *tCertPoolMultithreadingImpl) Stop() (err error) {
 	return
 }
 
-func (tCertPool *tCertPoolMultithreadingImpl) GetNextTCert() (tCert tCert, err error) {
+func (tCertPool *tCertPoolMultithreadingImpl) GetNextTCerts(nCerts int) (tCerts []tCert, err error) {
+	// The MT is considered WIP. We are going to add this support in the next round of refactoring, but added this
```

Comment:
That's an interesting one. Thanks for this.

So while `....pool_mt.go` file states at [line 24](https://github.com/hyperledger/fabric/blob/ca8135b1a0f95127c96478518628a65a377ec123/core/crypto/client_tcert_pool_mt.go#L24) that "_The Multi-threaded tCertPool is currently not used.  It plays only a role in testing._", and the internal (multi-threaded) test (MT/ST is a configurable setting)... I did make sure that it works at least for requesting 1 TCert - I did update to incorporate your comments and made sure it works. See the core MT test passing as well, please.